Library/Homebrew/requirements/x11_requirement.rb

```diff
@@ -17,32 +15,39 @@ def initialize(name = "x11", tags = [])
   end
 
   def min_version
-    # TODO: remove in https://github.com/Homebrew/brew/pull/3483
-    return Version::NULL unless OS.mac?
+    "1.12.2"
+  end
 
-    MacOS::XQuartz.minimum_version
+  def min_xdpyinfo_version
+    "1.3.0"
   end
 
   satisfy build_env: false do
-    # TODO: remove in https://github.com/Homebrew/brew/pull/3483
-    next false unless OS.mac?
-
-    next false unless MacOS::XQuartz.installed?
-    min_version <= MacOS::XQuartz.version
+    if which "Xorg"
+      version = Utils.popen_read "Xorg", "-version", err: :out
```

Comment:
@sjackman I think this needs to pass the return value of `which "Xorg"` to `Utils.popen_read`, otherwise when env filtering is enabled, it will end up with `command not found: Xorg`. Ditto `xdpyinfo` below.

Reply:
I believe that both `which` and `Utils.popen_read` will experience the same `PATH`. Either they'll both succeed, or they'll both fail. Do you believe that one of the two experiences env filtering, while the other does not? When `Xorg` and `xdpyinfo` are in the standard system path, both `which` and `Utils.popen_read` should work with env filtering enabled.

Reply:
>I believe that both which and Utils.popen_read will experience the same PATH. 

They will not. My comment was not speculative but based on testing. `which` is Homebrew's custom `which`, which brings back in the ORIGINAL_PATH.

Other requirements use Homebrew's `which` for this reason. There's no guarantee things used to satisfy requirements are in the env filtered PATH.

Reply:
Please point me to the relevant code.
`which` uses `PATH` 
https://github.com/Homebrew/brew/blob/master/Library/Homebrew/utils.rb#L309
`popen_read` uses `PATH`
https://github.com/Homebrew/brew/blob/master/Library/Homebrew/utils/popen.rb#L18
What you're saying may well be true, but it's not evident from the code.

Reply:
[Requirement#which](https://github.com/Homebrew/brew/blob/master/Library/Homebrew/requirement.rb#L158-L160) and [Requirment#which_all](https://github.com/Homebrew/brew/blob/master/Library/Homebrew/requirement.rb#L162-L164)

Reply:
Fixed. Thanks for the feedback, Joe.